[
  {
    "url": "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/288303520",
    "html_url": "https://github.com/bitcoin/bitcoin/pull/10015#issuecomment-288303520",
    "issue_url": "https://api.github.com/repos/bitcoin/bitcoin/issues/10015",
    "id": 288303520,
    "node_id": "MDEyOklzc3VlQ29tbWVudDI4ODMwMzUyMA==",
    "user": {
      "login": "NicolasDorier",
      "id": 3020646,
      "node_id": "MDQ6VXNlcjMwMjA2NDY=",
      "avatar_url": "https://avatars.githubusercontent.com/u/3020646?u=256d78207f88d303d61b62060a309657276cf6cb&v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/NicolasDorier",
      "html_url": "https://github.com/NicolasDorier",
      "followers_url": "https://api.github.com/users/NicolasDorier/followers",
      "following_url": "https://api.github.com/users/NicolasDorier/following{/other_user}",
      "gists_url": "https://api.github.com/users/NicolasDorier/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/NicolasDorier/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/NicolasDorier/subscriptions",
      "organizations_url": "https://api.github.com/users/NicolasDorier/orgs",
      "repos_url": "https://api.github.com/users/NicolasDorier/repos",
      "events_url": "https://api.github.com/users/NicolasDorier/events{/privacy}",
      "received_events_url": "https://api.github.com/users/NicolasDorier/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2017-03-22T05:37:21Z",
    "updated_at": "2017-03-22T05:37:21Z",
    "author_association": "CONTRIBUTOR",
    "body": "NACK on \"Coin Selection Failed\". It would break existing code (I rely in some project on the Insufficient Funds message), and I think \"Insufficient funds\" is correct. If you can't create a transaction because the chain is too long, then you really do not have enough liquidity.\r\n",
    "reactions": {
      "url": "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/288303520/reactions",
      "total_count": 0,
      "+1": 0,
      "-1": 0,
      "laugh": 0,
      "hooray": 0,
      "confused": 0,
      "heart": 0,
      "rocket": 0,
      "eyes": 0
    },
    "performed_via_github_app": null
  },
  {
    "url": "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/288387822",
    "html_url": "https://github.com/bitcoin/bitcoin/pull/10015#issuecomment-288387822",
    "issue_url": "https://api.github.com/repos/bitcoin/bitcoin/issues/10015",
    "id": 288387822,
    "node_id": "MDEyOklzc3VlQ29tbWVudDI4ODM4NzgyMg==",
    "user": {
      "login": "jnewbery",
      "id": 1063656,
      "node_id": "MDQ6VXNlcjEwNjM2NTY=",
      "avatar_url": "https://avatars.githubusercontent.com/u/1063656?u=71263a00d1269c7f0ed6660aa0c15c894df78a88&v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/jnewbery",
      "html_url": "https://github.com/jnewbery",
      "followers_url": "https://api.github.com/users/jnewbery/followers",
      "following_url": "https://api.github.com/users/jnewbery/following{/other_user}",
      "gists_url": "https://api.github.com/users/jnewbery/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/jnewbery/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/jnewbery/subscriptions",
      "organizations_url": "https://api.github.com/users/jnewbery/orgs",
      "repos_url": "https://api.github.com/users/jnewbery/repos",
      "events_url": "https://api.github.com/users/jnewbery/events{/privacy}",
      "received_events_url": "https://api.github.com/users/jnewbery/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2017-03-22T12:47:36Z",
    "updated_at": "2017-03-22T12:47:36Z",
    "author_association": "MEMBER",
    "body": "@NicolasDorier I strongly disagree with your definition of 'liquidity' here. In the too-long-chains case, the coin selection is being rejected by *local* policy. If the user was able to get the transaction to a miner who was prepared to mine a long chain, then it would be a perfectly valid transaction. I also think the most obvious reading of \"insufficient funds\" is that there are not enough funds in the wallet, not that the funds are temporarily unusable for technical reasons.\r\n\r\nThat said, I do have sympathy for client applications which expect the interface to bitcoind to remain stable. It's just that in this case the interface is wrong and should be corrected. If your client application is propagating the error message to the end user that there are insufficient funds, then it's likely to cause alarm and anxiety for users who may think that their funds have been lost.\r\n\r\nIdeally, we'd have error messages that were specific and correct, and that's where we should aim to get to. The choice here is between log messages that are (specific and potentially incorrect) and log messages that are (general and correct). Log messages that are general and correct are preferable.",
    "reactions": {
      "url": "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/288387822/reactions",
      "total_count": 0,
      "+1": 0,
      "-1": 0,
      "laugh": 0,
      "hooray": 0,
      "confused": 0,
      "heart": 0,
      "rocket": 0,
      "eyes": 0
    },
    "performed_via_github_app": null
  },
  {
    "url": "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/288596931",
    "html_url": "https://github.com/bitcoin/bitcoin/pull/10015#issuecomment-288596931",
    "issue_url": "https://api.github.com/repos/bitcoin/bitcoin/issues/10015",
    "id": 288596931,
    "node_id": "MDEyOklzc3VlQ29tbWVudDI4ODU5NjkzMQ==",
    "user": {
      "login": "NicolasDorier",
      "id": 3020646,
      "node_id": "MDQ6VXNlcjMwMjA2NDY=",
      "avatar_url": "https://avatars.githubusercontent.com/u/3020646?u=256d78207f88d303d61b62060a309657276cf6cb&v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/NicolasDorier",
      "html_url": "https://github.com/NicolasDorier",
      "followers_url": "https://api.github.com/users/NicolasDorier/followers",
      "following_url": "https://api.github.com/users/NicolasDorier/following{/other_user}",
      "gists_url": "https://api.github.com/users/NicolasDorier/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/NicolasDorier/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/NicolasDorier/subscriptions",
      "organizations_url": "https://api.github.com/users/NicolasDorier/orgs",
      "repos_url": "https://api.github.com/users/NicolasDorier/repos",
      "events_url": "https://api.github.com/users/NicolasDorier/events{/privacy}",
      "received_events_url": "https://api.github.com/users/NicolasDorier/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2017-03-23T02:13:25Z",
    "updated_at": "2017-03-23T02:13:25Z",
    "author_association": "CONTRIBUTOR",
    "body": "@jnewbery well in my case, the Insufficient Fee is interpreted into a proper Enum value in my client code.\r\nMaking this change mean that my code will throw a UnknownException, interpreted like a bug, instead of a Insufficient Fee, which delay the same action for later. (and if long chains, this is how it should be handled as well)\r\n\r\nThat said this PR should solve my issue https://github.com/bitcoin/bitcoin/issues/9874 .\r\nAt least \"Insufficient Fee\" so you do not break older code, and maybe add \"Too long chain\" as a new error message in the case that this PR attempt to fix.\r\n\r\n",
    "reactions": {
      "url": "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/288596931/reactions",
      "total_count": 0,
      "+1": 0,
      "-1": 0,
      "laugh": 0,
      "hooray": 0,
      "confused": 0,
      "heart": 0,
      "rocket": 0,
      "eyes": 0
    },
    "performed_via_github_app": null
  },
  {
    "url": "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/288597075",
    "html_url": "https://github.com/bitcoin/bitcoin/pull/10015#issuecomment-288597075",
    "issue_url": "https://api.github.com/repos/bitcoin/bitcoin/issues/10015",
    "id": 288597075,
    "node_id": "MDEyOklzc3VlQ29tbWVudDI4ODU5NzA3NQ==",
    "user": {
      "login": "NicolasDorier",
      "id": 3020646,
      "node_id": "MDQ6VXNlcjMwMjA2NDY=",
      "avatar_url": "https://avatars.githubusercontent.com/u/3020646?u=256d78207f88d303d61b62060a309657276cf6cb&v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/NicolasDorier",
      "html_url": "https://github.com/NicolasDorier",
      "followers_url": "https://api.github.com/users/NicolasDorier/followers",
      "following_url": "https://api.github.com/users/NicolasDorier/following{/other_user}",
      "gists_url": "https://api.github.com/users/NicolasDorier/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/NicolasDorier/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/NicolasDorier/subscriptions",
      "organizations_url": "https://api.github.com/users/NicolasDorier/orgs",
      "repos_url": "https://api.github.com/users/NicolasDorier/repos",
      "events_url": "https://api.github.com/users/NicolasDorier/events{/privacy}",
      "received_events_url": "https://api.github.com/users/NicolasDorier/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2017-03-23T02:14:31Z",
    "updated_at": "2017-03-23T02:15:11Z",
    "author_association": "CONTRIBUTOR",
    "body": "I advice \"too-long-mempool-chain\" as it is the same error message sent by `sendrawtransaction`, if this transaction was constructed.",
    "reactions": {
      "url": "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/288597075/reactions",
      "total_count": 0,
      "+1": 0,
      "-1": 0,
      "laugh": 0,
      "hooray": 0,
      "confused": 0,
      "heart": 0,
      "rocket": 0,
      "eyes": 0
    },
    "performed_via_github_app": null
  },
  {
    "url": "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/312340016",
    "html_url": "https://github.com/bitcoin/bitcoin/pull/10015#issuecomment-312340016",
    "issue_url": "https://api.github.com/repos/bitcoin/bitcoin/issues/10015",
    "id": 312340016,
    "node_id": "MDEyOklzc3VlQ29tbWVudDMxMjM0MDAxNg==",
    "user": {
      "login": "jnewbery",
      "id": 1063656,
      "node_id": "MDQ6VXNlcjEwNjM2NTY=",
      "avatar_url": "https://avatars.githubusercontent.com/u/1063656?u=71263a00d1269c7f0ed6660aa0c15c894df78a88&v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/jnewbery",
      "html_url": "https://github.com/jnewbery",
      "followers_url": "https://api.github.com/users/jnewbery/followers",
      "following_url": "https://api.github.com/users/jnewbery/following{/other_user}",
      "gists_url": "https://api.github.com/users/jnewbery/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/jnewbery/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/jnewbery/subscriptions",
      "organizations_url": "https://api.github.com/users/jnewbery/orgs",
      "repos_url": "https://api.github.com/users/jnewbery/repos",
      "events_url": "https://api.github.com/users/jnewbery/events{/privacy}",
      "received_events_url": "https://api.github.com/users/jnewbery/received_events",
      "type": "User",
      "site_admin": false
    },
    "created_at": "2017-06-30T18:24:40Z",
    "updated_at": "2017-06-30T18:24:40Z",
    "author_association": "MEMBER",
    "body": "Doesn't seem to be much appetite for this. There was a bit of disagreement on IRC here: https://botbot.me/freenode/bitcoin-core-dev/2017-03-16/?msg=82546050&page=3\r\n\r\nClosing for now.",
    "reactions": {
      "url": "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/312340016/reactions",
      "total_count": 0,
      "+1": 0,
      "-1": 0,
      "laugh": 0,
      "hooray": 0,
      "confused": 0,
      "heart": 0,
      "rocket": 0,
      "eyes": 0
    },
    "performed_via_github_app": null
  },
  {
    "url": "https://api.github.com/repos/bitcoin/bitcoin/pulls/comments/106509226",
    "pull_request_review_id": 27435440,
    "id": 106509226,
    "node_id": "MDI0OlB1bGxSZXF1ZXN0UmV2aWV3Q29tbWVudDEwNjUwOTIyNg==",
    "diff_hunk": "@@ -316,7 +316,7 @@ def run_test(self):\n         rawtx   = self.nodes[2].createrawtransaction(inputs, outputs)\n         dec_tx  = self.nodes[2].decoderawtransaction(rawtx)\n \n-        assert_raises_jsonrpc(-4, \"Insufficient funds\", self.nodes[2].fundrawtransaction, rawtx)\n+        assert_raises_jsonrpc(-4, \"Coin Selection Failed\", self.nodes[2].fundrawtransaction, rawtx)",
    "path": "qa/rpc-tests/fundrawtransaction.py",
    "position": 5,
    "original_position": 5,
    "commit_id": "2d825c8582a5b6e9a78e48776b5ec3240141b050",
    "original_commit_id": "97cb164f54f48f9c91293d08f6afc7d5628dc789",
    "user": {
      "login": "lontivero",
      "id": 127973,
      "node_id": "MDQ6VXNlcjEyNzk3Mw==",
      "avatar_url": "https://avatars.githubusercontent.com/u/127973?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/lontivero",
      "html_url": "https://github.com/lontivero",
      "followers_url": "https://api.github.com/users/lontivero/followers",
      "following_url": "https://api.github.com/users/lontivero/following{/other_user}",
      "gists_url": "https://api.github.com/users/lontivero/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/lontivero/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/lontivero/subscriptions",
      "organizations_url": "https://api.github.com/users/lontivero/orgs",
      "repos_url": "https://api.github.com/users/lontivero/repos",
      "events_url": "https://api.github.com/users/lontivero/events{/privacy}",
      "received_events_url": "https://api.github.com/users/lontivero/received_events",
      "type": "User",
      "site_admin": false
    },
    "body": "The new error message is a bit confussing to me because it doesn't provides a reason or error code. IMO if something can fail for two different reasons, it should let the caller know the reason in order to let him take the appropiate actions, show the best error message, etc.",
    "created_at": "2017-03-16T19:22:32Z",
    "updated_at": "2017-03-16T19:22:32Z",
    "html_url": "https://github.com/bitcoin/bitcoin/pull/10015#discussion_r106509226",
    "pull_request_url": "https://api.github.com/repos/bitcoin/bitcoin/pulls/10015",
    "author_association": "CONTRIBUTOR",
    "_links": {
      "self": {
        "href": "https://api.github.com/repos/bitcoin/bitcoin/pulls/comments/106509226"
      },
      "html": {
        "href": "https://github.com/bitcoin/bitcoin/pull/10015#discussion_r106509226"
      },
      "pull_request": {
        "href": "https://api.github.com/repos/bitcoin/bitcoin/pulls/10015"
      }
    },
    "reactions": {
      "url": "https://api.github.com/repos/bitcoin/bitcoin/pulls/comments/106509226/reactions",
      "total_count": 0,
      "+1": 0,
      "-1": 0,
      "laugh": 0,
      "hooray": 0,
      "confused": 0,
      "heart": 0,
      "rocket": 0,
      "eyes": 0
    },
    "start_line": null,
    "original_start_line": null,
    "start_side": null,
    "line": 319,
    "original_line": 319,
    "side": "RIGHT"
  },
  {
    "url": "https://api.github.com/repos/bitcoin/bitcoin/pulls/comments/106511904",
    "pull_request_review_id": 27438409,
    "id": 106511904,
    "node_id": "MDI0OlB1bGxSZXF1ZXN0UmV2aWV3Q29tbWVudDEwNjUxMTkwNA==",
    "diff_hunk": "@@ -316,7 +316,7 @@ def run_test(self):\n         rawtx   = self.nodes[2].createrawtransaction(inputs, outputs)\n         dec_tx  = self.nodes[2].decoderawtransaction(rawtx)\n \n-        assert_raises_jsonrpc(-4, \"Insufficient funds\", self.nodes[2].fundrawtransaction, rawtx)\n+        assert_raises_jsonrpc(-4, \"Coin Selection Failed\", self.nodes[2].fundrawtransaction, rawtx)",
    "path": "qa/rpc-tests/fundrawtransaction.py",
    "position": 5,
    "original_position": 5,
    "commit_id": "2d825c8582a5b6e9a78e48776b5ec3240141b050",
    "original_commit_id": "97cb164f54f48f9c91293d08f6afc7d5628dc789",
    "user": {
      "login": "jnewbery",
      "id": 1063656,
      "node_id": "MDQ6VXNlcjEwNjM2NTY=",
      "avatar_url": "https://avatars.githubusercontent.com/u/1063656?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/jnewbery",
      "html_url": "https://github.com/jnewbery",
      "followers_url": "https://api.github.com/users/jnewbery/followers",
      "following_url": "https://api.github.com/users/jnewbery/following{/other_user}",
      "gists_url": "https://api.github.com/users/jnewbery/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/jnewbery/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/jnewbery/subscriptions",
      "organizations_url": "https://api.github.com/users/jnewbery/orgs",
      "repos_url": "https://api.github.com/users/jnewbery/repos",
      "events_url": "https://api.github.com/users/jnewbery/events{/privacy}",
      "received_events_url": "https://api.github.com/users/jnewbery/received_events",
      "type": "User",
      "site_admin": false
    },
    "body": "I agree it'd be better to have more precise error codes/messages returned. However, I think this is an improvement since we're returning \"Insufficient funds\" when the problem may be something different.",
    "created_at": "2017-03-16T19:35:19Z",
    "updated_at": "2017-03-16T19:35:19Z",
    "html_url": "https://github.com/bitcoin/bitcoin/pull/10015#discussion_r106511904",
    "pull_request_url": "https://api.github.com/repos/bitcoin/bitcoin/pulls/10015",
    "author_association": "MEMBER",
    "_links": {
      "self": {
        "href": "https://api.github.com/repos/bitcoin/bitcoin/pulls/comments/106511904"
      },
      "html": {
        "href": "https://github.com/bitcoin/bitcoin/pull/10015#discussion_r106511904"
      },
      "pull_request": {
        "href": "https://api.github.com/repos/bitcoin/bitcoin/pulls/10015"
      }
    },
    "reactions": {
      "url": "https://api.github.com/repos/bitcoin/bitcoin/pulls/comments/106511904/reactions",
      "total_count": 0,
      "+1": 0,
      "-1": 0,
      "laugh": 0,
      "hooray": 0,
      "confused": 0,
      "heart": 0,
      "rocket": 0,
      "eyes": 0
    },
    "start_line": null,
    "original_start_line": null,
    "start_side": null,
    "line": 319,
    "original_line": 319,
    "side": "RIGHT",
    "in_reply_to_id": 106509226
  },
  {
    "url": "https://api.github.com/repos/bitcoin/bitcoin/pulls/comments/106623497",
    "pull_request_review_id": 27554868,
    "id": 106623497,
    "node_id": "MDI0OlB1bGxSZXF1ZXN0UmV2aWV3Q29tbWVudDEwNjYyMzQ5Nw==",
    "diff_hunk": "@@ -316,7 +316,7 @@ def run_test(self):\n         rawtx   = self.nodes[2].createrawtransaction(inputs, outputs)\n         dec_tx  = self.nodes[2].decoderawtransaction(rawtx)\n \n-        assert_raises_jsonrpc(-4, \"Insufficient funds\", self.nodes[2].fundrawtransaction, rawtx)\n+        assert_raises_jsonrpc(-4, \"Coin Selection Failed\", self.nodes[2].fundrawtransaction, rawtx)",
    "path": "qa/rpc-tests/fundrawtransaction.py",
    "position": 5,
    "original_position": 5,
    "commit_id": "2d825c8582a5b6e9a78e48776b5ec3240141b050",
    "original_commit_id": "97cb164f54f48f9c91293d08f6afc7d5628dc789",
    "user": {
      "login": "laanwj",
      "id": 126646,
      "node_id": "MDQ6VXNlcjEyNjY0Ng==",
      "avatar_url": "https://avatars.githubusercontent.com/u/126646?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/laanwj",
      "html_url": "https://github.com/laanwj",
      "followers_url": "https://api.github.com/users/laanwj/followers",
      "following_url": "https://api.github.com/users/laanwj/following{/other_user}",
      "gists_url": "https://api.github.com/users/laanwj/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/laanwj/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/laanwj/subscriptions",
      "organizations_url": "https://api.github.com/users/laanwj/orgs",
      "repos_url": "https://api.github.com/users/laanwj/repos",
      "events_url": "https://api.github.com/users/laanwj/events{/privacy}",
      "received_events_url": "https://api.github.com/users/laanwj/received_events",
      "type": "User",
      "site_admin": false
    },
    "body": "Agree that if there are really insufficient funds it'd be good to reply that. A specific error is better than a non-specific one *if the message is correct*. Better an aspecific error than an incorrect one though. An incorrect \"insufficient funds\" error could set the caller on a wild goose chase or get them to panic. In that case it's better to return something general which prompts the caller to do their own research.\r\n\r\n\r\n",
    "created_at": "2017-03-17T10:38:18Z",
    "updated_at": "2017-03-17T10:38:18Z",
    "html_url": "https://github.com/bitcoin/bitcoin/pull/10015#discussion_r106623497",
    "pull_request_url": "https://api.github.com/repos/bitcoin/bitcoin/pulls/10015",
    "author_association": "MEMBER",
    "_links": {
      "self": {
        "href": "https://api.github.com/repos/bitcoin/bitcoin/pulls/comments/106623497"
      },
      "html": {
        "href": "https://github.com/bitcoin/bitcoin/pull/10015#discussion_r106623497"
      },
      "pull_request": {
        "href": "https://api.github.com/repos/bitcoin/bitcoin/pulls/10015"
      }
    },
    "reactions": {
      "url": "https://api.github.com/repos/bitcoin/bitcoin/pulls/comments/106623497/reactions",
      "total_count": 0,
      "+1": 0,
      "-1": 0,
      "laugh": 0,
      "hooray": 0,
      "confused": 0,
      "heart": 0,
      "rocket": 0,
      "eyes": 0
    },
    "start_line": null,
    "original_start_line": null,
    "start_side": null,
    "line": 319,
    "original_line": 319,
    "side": "RIGHT",
    "in_reply_to_id": 106509226
  },
  {
    "url": "https://api.github.com/repos/bitcoin/bitcoin/pulls/comments/107335949",
    "pull_request_review_id": 28299869,
    "id": 107335949,
    "node_id": "MDI0OlB1bGxSZXF1ZXN0UmV2aWV3Q29tbWVudDEwNzMzNTk0OQ==",
    "diff_hunk": "@@ -316,7 +316,7 @@ def run_test(self):\n         rawtx   = self.nodes[2].createrawtransaction(inputs, outputs)\n         dec_tx  = self.nodes[2].decoderawtransaction(rawtx)\n \n-        assert_raises_jsonrpc(-4, \"Insufficient funds\", self.nodes[2].fundrawtransaction, rawtx)\n+        assert_raises_jsonrpc(-4, \"Coin Selection Failed\", self.nodes[2].fundrawtransaction, rawtx)",
    "path": "qa/rpc-tests/fundrawtransaction.py",
    "position": 5,
    "original_position": 5,
    "commit_id": "2d825c8582a5b6e9a78e48776b5ec3240141b050",
    "original_commit_id": "97cb164f54f48f9c91293d08f6afc7d5628dc789",
    "user": {
      "login": "NicolasDorier",
      "id": 3020646,
      "node_id": "MDQ6VXNlcjMwMjA2NDY=",
      "avatar_url": "https://avatars.githubusercontent.com/u/3020646?v=4",
      "gravatar_id": "",
      "url": "https://api.github.com/users/NicolasDorier",
      "html_url": "https://github.com/NicolasDorier",
      "followers_url": "https://api.github.com/users/NicolasDorier/followers",
      "following_url": "https://api.github.com/users/NicolasDorier/following{/other_user}",
      "gists_url": "https://api.github.com/users/NicolasDorier/gists{/gist_id}",
      "starred_url": "https://api.github.com/users/NicolasDorier/starred{/owner}{/repo}",
      "subscriptions_url": "https://api.github.com/users/NicolasDorier/subscriptions",
      "organizations_url": "https://api.github.com/users/NicolasDorier/orgs",
      "repos_url": "https://api.github.com/users/NicolasDorier/repos",
      "events_url": "https://api.github.com/users/NicolasDorier/events{/privacy}",
      "received_events_url": "https://api.github.com/users/NicolasDorier/received_events",
      "type": "User",
      "site_admin": false
    },
    "body": "I think \"Insufficient funds\" is correct. If the user does not have enough liquidity to not do a long chain, then he does not have enough funds.",
    "created_at": "2017-03-22T05:38:35Z",
    "updated_at": "2017-03-22T05:38:35Z",
    "html_url": "https://github.com/bitcoin/bitcoin/pull/10015#discussion_r107335949",
    "pull_request_url": "https://api.github.com/repos/bitcoin/bitcoin/pulls/10015",
    "author_association": "CONTRIBUTOR",
    "_links": {
      "self": {
        "href": "https://api.github.com/repos/bitcoin/bitcoin/pulls/comments/107335949"
      },
      "html": {
        "href": "https://github.com/bitcoin/bitcoin/pull/10015#discussion_r107335949"
      },
      "pull_request": {
        "href": "https://api.github.com/repos/bitcoin/bitcoin/pulls/10015"
      }
    },
    "reactions": {
      "url": "https://api.github.com/repos/bitcoin/bitcoin/pulls/comments/107335949/reactions",
      "total_count": 0,
      "+1": 0,
      "-1": 0,
      "laugh": 0,
      "hooray": 0,
      "confused": 0,
      "heart": 0,
      "rocket": 0,
      "eyes": 0
    },
    "start_line": null,
    "original_start_line": null,
    "start_side": null,
    "line": 319,
    "original_line": 319,
    "side": "RIGHT",
    "in_reply_to_id": 106509226
  }
]